hxrsnd/macromotor.py

```diff
@@ -3,46 +3,49 @@
 
 All units of time are in picoseconds, units of length are in mm.
 """
-############
-# Standard #
-############
+
 import os
+import time
 import logging
-
-###############
-# Third Party #
-###############
 import numpy as np
 import pandas as pd
 from ophyd.utils import LimitError
 from ophyd.status import wait as status_wait
-
-########
-# SLAC #
-########
 from pcdsdevices.device import Device
-
-##########
-# Module #
-##########
+from pcdsdevices.signal import Signal
+from pcdsdevices.component import Component
 from .utils import as_list, flatten
 from .bragg import bragg_angle, cosd, sind
 from .exceptions import MotorDisabled, MotorFaulted, MotorStopped, BadN2Pressure
 
 logger = logging.getLogger(__name__)
 
 
+class PythonSignal(Signal):
+    """
+    Signal that returns the results of a python function from the get method.
+    """
+    def __init__(self, func, *args, **kwargs):
+        self._func = func
+        super().__init__(*args, **kwargs)
+        
+    def get(self):
+        return self._func()
```

Comment:
This class is only subtly different than the built-in ophyd.sim.SynSignal

Reply:
Hmm, this is true. I'll go through and see if this is even necessary then.

Reply:
Is requiring a `trigger` call for the signal value to update make sense for motor parameters? I'm a little put off by it, but I'm not sure if this is how signals are supposed to work by design.

Reply:
It's probably built on an assumption that if you care enough about a function to make it a Signal, it's probably kind of slow and you don't want to call it many times. I think it's totally good to use this class here, just wanted to make sure you took a step back and made sure you weren't duplicating effort.

Reply:
ah okay, good looks. I totally missed this as a use case for it.

Reply:
Actually, since this is literally just so be able to list a `read_attr`, would it be in bad form to just use an empty `Signal` instead of returning something that's redundant at best?